scala/scala.bzl

```diff
@@ -599,9 +650,14 @@ def _scala_binary_common(ctx, cjars, rjars, transitive_compile_time_jars, jars2l
       class_jar=outputs.class_jar,
       deploy_jar=ctx.outputs.deploy_jar,
   )
+
+  compile_outputs = [outputs.class_jar]
+  if outputs.java_jar:
+    compile_outputs.extend([outputs.java_jar.ijar])
+
   scalaattr = struct(
       outputs = rule_outputs,
-      compile_jars = depset([outputs.class_jar]),
```

Comment:
@johnynek is there any reason you use here `outputs.class_jar` and in `_lib` you use `ctx.outputs.jar`?
I did a few prints and they seem identical.

Reply:
@johnynek ping

Reply:
I don't think there was a plan. I think it probably was just ill-unified.

Reply:
yeah, it looks like `outputs.class_jar` is always `ctx.outputs.jar` in all cases. We could probably remove it and only use `ctx.outputs.jar` to be more clear.

Reply:
Decided to leave it for now. Since the outputs.class_jar (although the unclear name) adds symmetry to the collection of the java jar.
We'll handle it when we'll do a bigger refactoring and re-alignment